### ISSUE TITLE
No output for certain input blocks in static mode

### ISSUE THREAD

**Issue by saranyuc3** [tomtor/HDL-deflate#5]
Starting from reset, I send the following input block in static mode, 0x18 0x19 0x72 0x04 0x00  (which is compression of 'A'), then set imode == DECOMPRESS in next cycle and imode=IDLE for all cycles onwards, the o_oprogress never increments. When I send three blocks  0x18 0x19 0x72 0x04 0xC8 0x09 0x04 0x67 0x40 0x00 (one block for each 'A', 'B' and 'C'), and then decompress, the output is 0x41 ('A'), 0x42 ('B') and output for 'C' is never produced.

**Comment by tomtor** [comment]
How did you test this?

Can you reproduce the error in a (new) python test and share that code?

**Comment by saranyuc3** [comment]
I tested this on the RTL. Does an RTL testbench work for you? It's ''probably'' working correctly for the python code.

**Comment by tomtor** [comment]
> I tested this on the RTL. Does an RTL testbench work for you? It's ''probably'' working correctly for the python code.

I tested it on real hardware, see README.

Also the generated RTL bench works. I assume you don't correctly write the last input byte and the conversion signals it is done.

**Comment by saranyuc3** [comment]
Agreed that the last byte was wrong. Instead of 0x00 it should be 0x60 (now it incorporates a part of the header for the next byte). Now the input is 0x18 0x19 0x72 0x04 0x60.  But the problem remains the same.
